addons/l10n_br/models/res_partner_bank.py

```diff
@@ -80,7 +80,8 @@ def _get_qr_code_vals_list(self, *args, **kwargs):
         Although not specified explicitly in the spec, not uppercasing causes errors when scanning the code."""
         res = super()._get_qr_code_vals_list(*args, **kwargs)
         if self.country_code == "BR":
-            res[5] = (res[5][0], float_repr(res[5][1], 2))  # amount
+            if res[5][1] is not None:
+                res[5] = (res[5][0], float_repr(res[5][1], 2))  # amount
```

Comment:
Just to be sure, in that case we should keep `(54, None) `and not something else like `(54, 0.0)` ?

Reply:
That's a good question, I'm working on a separate issue where [not having the reference field breaks things for some bank apps](https://www.odoo.com/web#id=3818534&cids=3&model=project.task&view_type=form). This passes [validation](https://pix.nascent.com.br/) [tools](https://openpix.com.br/qrcode/scanner/), but so did not having a reference. I just asked our Brazilian PO to test out these Pix codes without an amount with her arsenal of bank apps, let's see what she says :).

Reply:
Alright, can you re-request a review when you have more info ? Thx :)

Reply:
Will do! We'll probably know early next week.

Reply:
FYI, it's been tested and confirmed working. However, in the meantime https://github.com/odoo/odoo/pull/159020 was opened as well. I'll let them know.